website/blog/2025-04-08-run-ferretdb-postgres-documentdb-extension-cnpg-kubernetes.md

```diff
@@ -0,0 +1,243 @@
+
+  bootstrap:
+    initdb:
+      postInitSQL:
+        - 'CREATE EXTENSION IF NOT EXISTS documentdb CASCADE;'
+```
+
+We explicitly enable `enableSuperuserAccess` so that we can connect with the default `postgres` user.
+
+CloudNativePG runs PostgreSQL with UID 26 by default, but the DocumentDB image requires UID and GID 999.
+Be sure to set both explicitly to avoid permission errors.
+
+We also load the required extensions and enable shared preload libraries needed by FerretDB.
+The `cron.database_name` parameter is set to `postgres` to ensure that the cron jobs run in the correct database.
+
+The `bootstrap` section is where we run the SQL command to create the DocumentDB extension.
+Since CloudNativePG doesn't execute entrypoint scripts from the DocumentDB image, we need to manually create the extension during initialization.
+
+Apply it:
+
+```sh
+kubectl apply -f pg-cluster.yaml -n cnpg
+```
+
+CNPG will handle cluster creation, persistent storage, and generate a password for the `postgres` superuser in Secret.
+
+You can check the status of the cluster with:
+
+```sh
+kubectl get cluster -n cnpg
+```
+
+You should see the cluster in `Running` state.
+
+```text
+NAME               AGE    INSTANCES   READY   STATUS                     PRIMARY
+postgres-cluster   176m   3           3       Cluster in healthy state   postgres-cluster-1
+```
+
+To get the generated password for the `postgres` user:
+
+```sh
+kubectl get secret -n cnpg postgres-cluster-superuser -o jsonpath='{.data.password}' | base64 -d && echo
+```
+
+## Deploy FerretDB
+
+With the PostgreSQL backend ready, let's deploy FerretDB.
+FerretDB expects you to specify the connection string to the PostgreSQL backend via the `FERRETDB_POSTGRESQL_URL` environment variable.
+
+Save this as `ferretdb.yaml`:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ferretdb
+  namespace: cnpg
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ferretdb
+  template:
+    metadata:
+      labels:
+        app: ferretdb
+    spec:
+      containers:
+        - name: ferretdb
+          image: ghcr.io/ferretdb/ferretdb:2.1.0
+          ports:
+            - containerPort: 27017
+          env:
+            - name: FERRETDB_POSTGRESQL_URL
+              value: 'postgresql://postgres:<password>@postgres-cluster-rw.cnpg.svc.cluster.local:5432/postgres'
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ferretdb-service
+  namespace: cnpg
+spec:
+  selector:
+    app: ferretdb
+  ports:
+    - protocol: TCP
+      port: 27017
+      targetPort: 27017
+  type: NodePort
+```
+
+The connection string uses the `postgres` superuser created by CNPG, the service endpoint exposed by the PostgreSQL cluster (`postgres-cluster-rw`), and the default `postgres` database.
+Be sure to replace `<password>` with the actual password you retrieved earlier.
+For production environments, instead of hardcoding the password, consider using `Secrets` to manage sensitive information.
```

Comment:
```suggestion
For production environments, instead of hardcoding the password, consider using secrets to manage sensitive information.
```

There is no "`Secrets`"